fenicsadapter/adapter_core.py

```diff
@@ -97,16 +97,34 @@ def convert_fenics_to_precice(function, coupling_subdomain, fenics_dimensions, d
     array : array_like
         Array of FEniCS function values at each point on the boundary.
     """
+
+    if type(function) is not dolfin.Function:
+        raise Exception("Cannot handle data type {}".format(type(function)))
+
+    if function.function_space().num_sub_spaces() > 0:  # function space is a VectorFunctionSpace
+        linear_space = VectorFunctionSpace(function.function_space().mesh(), "P", 1)
+    else:
+        linear_space = FunctionSpace(function.function_space().mesh(), "P", 1)
+
     _, _, _, _, local_ids, _ = get_coupling_boundary_vertices(function.function_space().mesh(), function.function_space(),
                                                               coupling_subdomain, fenics_dimensions, dimensions)
```

Comment:
```suggestion
    _, _, _, _, local_ids, _ = get_coupling_boundary_vertices(function.function_space(), coupling_subdomain, fenics_dimensions, dimensions)
```

Here, we can avoid duplicate arguments by only providing the function space. This makes the function easier to use.